src/EnergyPlus/OutputProcessor.cc

```diff
@@ -7541,7 +7542,8 @@ void GetMeteredVariables(EnergyPlusData &state,
             TimeStepTypes(NumVariables) = op->RVariableTypes(Loop).timeStepType;
             unitsForVar(NumVariables) = op->RVariableTypes(Loop).units;
 
-            ResourceTypes.at(NumVariables) = AssignResourceTypeNum(UtilityRoutines::MakeUPPERCase(op->EnergyMeters(MeterPtr).ResourceType));
+            ResourceTypes.at(NumVariables) = static_cast<DataGlobalConstants::ResourceType>(
+                getEnumerationValue(DataGlobalConstants::ResourceTypeNamesUC, UtilityRoutines::MakeUPPERCase(op->EnergyMeters(MeterPtr).ResourceType)));
```

Comment:
If these are the only changes in OutputProcessor, then it seems like this PR breaks the possibility of reporting pollutants as meters. I'll make a test file to confirm. Oh, I see, [`GetStandardMeterResourceType`](https://github.com/NREL/EnergyPlus/blob/7cc2fa5a6f00d9f5db1cc54d853f11ea74851582/src/EnergyPlus/OutputProcessor.cc#L1515-L1541) has not been touched so it does not use `DataGlobalConstants::ResourceTypes` to validate yet. So, when it is changed over someday, it will have to check 2 enums, `ResourceTypes` and `PollutionsModule::Pollutant` (should that be `PollutantTypes`?) to cover all of the valid meter resources. But that's where this may fall apart, when the resource type key in `SetUpOutputVariable` is changed to an enum, that enum will need to include the union of `ResourceType` and `PollutantTypes`.

Reply:
Is this the *only* reason that these disparate enums are thrown together? To simplify OutputProcessor? That was a sub-optimal design choice. It's better to complicate OutputProcessor and simplify the remaining 98% of the code than the reverse.

Reply:
I will point out that in OutputProcessor specifically, ResourceType is a `string`, not an `enum`. That's not great in general but at least in this case it makes it simpler to deal with multiple enums as a single set in a compartmentalized way. The next question is whether the `ResourceType enum` is used anywhere other than as `FuelType`. From this PR, it looks like the answer is no.

Reply:
I don't claim to understand fully how OutputProcessor is managing meters and submeters, but the string name is important because that ultimately becomes part of the meter name like "Electricity:Facility". Ultimately I believe each metered value ends up being the sum of a list of pointers to the respective variables that feed into it, so at that point it doesn't care about ResourceType anymore, that's just a part of the particular meter's name.

Ultimately, when calling `SetupOutputVariable` I think we would prefer that `ResourceTypeKey` was an enum rather than a string. That's where the full combined enum would be needed, and then it would be promptly used to grab the string version to create the concatenated meter name. After that, I'm not sure if OutputProcessor usee the resource type explicitly or not.

Reply:
In that case, how would you suggest we handle the myriad cases where we only need the `FuelType` subset and the one case where we only need the `Pollutant` subset? Use separate enumerations for those and use `std::array`s to map one to the other?

Reply:
Maybe. `Pollutants` are a different beast, because all of the pollutants are active at the same time and the enum could be used to cycle through arrays by pollutant. So I think you really need a dedicated enum for the pollution module, but then, yes you want a way to look up the equivalent `ResourceType` enum value to pass into `SetupOutputVariable`.

`FuelType` on the other hand is used differently. Any given component, like a boiler, has a single main fuel type, so here, looking it up and storing it as a `ResourceType` is good, because it's only used to set up meters and name output variables. So, for this, perhaps there could be a companion bool array, say `IsValidFuelType` which is true or false for each ResourceType and then after `getenumeration` returns the `ResourceType` check 
```
if (IsValueFuelType[static_cast<int>Boiler.ResourceType])
```
or something like that?

Reply:
OutputReportTabular defines `numResourceTypes (14)` and `numSourceTypes (12)` as the first 14 and 12 items in `ResourceType` (the difference between the two is that `SourceTypes` does not include district heating and cooling). The former especially is used as in the index in a large number of arrays. How would you handle this use?

Reply:
> `FuelType` on the other hand is used differently. Any given component, like a boiler, has a single main fuel type, so here, looking it up and storing it as a `ResourceType` is good, because it's only used to set up meters and name output variables. So, for this, perhaps there could be a companion bool array, say `IsValidFuelType` which is true or false for each ResourceType and then after `getenumeration` returns the `ResourceType` check.
> 
> ```
> if (IsValueFuelType[static_cast<int>Boiler.ResourceType])
> ```
> 
> or something like that?

That's a possibility, and I grant you that there are different kinds of beasts. But `FuelType` is also used as array indices (or at least it should be instead of what is used now which is just the numbers 1, 2, 3, etc.) and there is something to be said for having a consistent way of doing things. I have an idea, I will talk to @dareumnam about it and we will see how it works.